nx/source/services/news.c

```diff
@@ -0,0 +1,308 @@
+#define NX_SERVICE_ASSUME_NON_DOMAIN
+#include "services/news.h"
+#include "service_guard.h"
+#include "runtime/hosversion.h"
+#include <string.h>
+
+static NewsServiceType g_newsServiceType;
+static Service g_newsCreatorSrv;
+static Service g_newsSrv;
+
+static Result _newsCreateNewsService(Service *srv_out);
+
+NX_GENERATE_SERVICE_GUARD_PARAMS(news, (NewsServiceType service_type), (service_type));
+
+Result _newsInitialize(NewsServiceType service_type) {
+    Result rc;
+
+    if (service_type > NewsServiceType_v)
```

Comment:
Have NewsServiceType_Count, do >= ServiceType_Count here instead of implicitly having _Viewer as the last service type in the enum.